gnocchiclient/exceptions.py

```diff
@@ -99,7 +99,7 @@ class NotFound(ClientException):
 
 class MetricNotFound(NotFound):
     message = "Metric not found"
-    match = re.compile("Metric .* does not exist")
+    match = re.compile("Metric .* does not exist|Metrics not found")
```

Comment:
I understand this change but I think the actual string there `Metrics not found` is wrong here [1] introduced in [2].

So the first string matches gnocchi.storage.MetricDoesNotExist what I don't understand is why when Gnocchi raises gnocchi.indexer.NoSuchMetric it doesn't become "Metrics <metric> does not exist" as show in code [3]

To me this feels like a bug in Gnocchi where we dont raise, catch or reraise something properly and we should fix that instead of catching the "cause" in the error message.

So instead of adding the "cause" here and trying to parse "cause" below in from_response(), we should figure out why gnocchi.indexer.NoSuchMetric (or where that is catched/reraised into an API response) doesn't give us "Metric <metric> does not exist".

If we want we can also fix that spelling error in [1] from "Metrics not found" -> "Metric not found" but that is unrelated this change.

What do you think?

[1] https://github.com/gnocchixyz/gnocchi/blob/master/gnocchi/indexer/__init__.py#L138
[2] https://github.com/gnocchixyz/gnocchi/issues/1013 
[3] https://github.com/gnocchixyz/gnocchi/blob/master/gnocchi/indexer/__init__.py#L132

Reply:
I see what you mean. It seems that there are two problems then. One, which is on Gnocchi server side not building the proper message, and the other in the client, that is using the description attribute as a string, when it is no. I will amend the code here and there then.